tests/unit/test_strike_service.py

```diff
@@ -0,0 +1,178 @@
+import pytest
+from pytest_mock.plugin import MockerFixture
+from moddingway.database.models import Strike
+from moddingway.services import strike_service, ban_service, exile_service
+from moddingway import enums
+from typing import List
+import discord
+
+
+@pytest.mark.parametrize(
+    "previous_points,total_points,expected_punishment",
+    [
+        (0, 1, "Nothing"),
+        (1, 8, "11 day exile"),
+        (8, 10, "14 day exile"),
+        (10, 11, "14 day exile"),
+        (1, 9, "11 day exile"),
+        (9, 12, "14 day exile"),
+        (12, 13, "14 day exile"),
+        (10, 15, "Permanent ban"),
+        (0, 3, "1 day exile"),
+        (10, 17, "Permanent ban"),
+    ],
+)
+@pytest.mark.asyncio
+async def test_apply_punisment(
+    previous_points: int,
+    total_points: int,
+    expected_punishment: str,
+    mocker: MockerFixture,
+    create_db_user,
+):
+    mocked_user = mocker.Mock()
+    mocked_db_user = create_db_user(get_strike_points=total_points)
+    mocked_ban = mocker.patch("moddingway.services.ban_service.ban_user")
+    mocked_embed = mocker.Mock(spec=discord.Embed)
+    # I tried to use patch here but it would override all test cases apparently spy can just observe without interrupting.
+    spy_calculate_punishment = mocker.spy(strike_service, "_calculate_punishment")
+
+    mocked_exile_user = mocker.patch("moddingway.services.exile_service.exile_user")
+
+    res = await strike_service._apply_punishment(
+        mocked_embed, mocked_user, mocked_db_user, previous_points
+    )
+    if total_points >= 15:
+        mocked_ban.assert_called_once_with(
+            mocked_user,
+            "Your strike were severe or frequent to be removed from NA Ultimate Raiding - FFXIV",
+            False,
+        )
+    else:
+        mocked_ban.assert_not_called()
+        spy_calculate_punishment.assert_called_once_with(previous_points, total_points)
+        punishment_days = spy_calculate_punishment.spy_return
+        if punishment_days == 0:
+            mocked_exile_user.assert_not_called()
+        else:
+            mocked_exile_user.assert_called_once_with(
+                mocked_embed,
+                mocked_user,
+                mocker.ANY,
+                "Your actions were severe or frequent enough for you to receive this exile",
```

Comment:
I don't think we need to assert on the user-facing text, this can be `mocker.Any`